.github/workflows/test.yaml

```diff
@@ -43,3 +43,18 @@ jobs:
       date: ${{ inputs.date }}
       sha: ${{ inputs.sha }}
       test_script: "ci/test_python_dask.sh"
+  wheel-tests-cuml:
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux-test.yml@upkeep/manylinux-improvements
+    with:
+      build_type: nightly
+      repo: rapidsai/cuml
+      branch: ${{ inputs.branch }}
+      date: ${{ inputs.date }}
+      sha: ${{ inputs.sha }}
+      package-name: cuml
+      test-before-amd64: "pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.02"
+      # On arm also need to install cupy from the specific webpage and CMake
+      # because treelite needs to be compiled (no wheels available for arm).
+      test-before-arm64: "pip install cupy-cuda11x -f https://pip.cupy.dev/aarch64 && pip install cmake && pip install git+https://github.com/dask/dask.git@main git+https://github.com/dask/distributed.git@main git+https://github.com/rapidsai/dask-cuda.git@branch-23.02"
+      test-unittest: "pytest -v ./python/cuml/tests -k 'not test_silhouette_score_batched'"
```

Comment:
Now that https://github.com/rapidsai/cuml/issues/5027 is resolved we can speed up the CI test runs with parallelism :tada: we need to avoid parallelizing dask tests though, so we'll need to do something like 
```suggestion
      test-unittest: "pytest -v ./python/cuml/tests -k 'not test_silhouette_score_batched' -n 8 --ignore=dask && pytest -v ./python/cuml/tests/dask"
```
I picked 8 procs since that's what the conda CI is doing, but if the executors for us have different available resources that may need to change.